File: app/src/main/java/me/zhaoweihao/hnuplus/Adapter/CommentAdapter.java
package me.zhaoweihao.hnuplus.Adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import me.zhaoweihao.hnuplus.JavaBean.Comment;
import me.zhaoweihao.hnuplus.R;

/**
 * Created by Administrator on 2017/11/10.
 */

public class CommentAdapter extends RecyclerView.Adapter<CommentAdapter.ViewHolder> {

    private List<Comment> mCommentList;

    static class ViewHolder extends RecyclerView.ViewHolder{

        TextView commentatorName,commentatorContent;

        public ViewHolder(View itemView) {

            super(itemView);
            commentatorName = (TextView) itemView.findViewById(R.id.tv_commentator_name);
            commentatorContent = (TextView) itemView.findViewById(R.id.tv_commentator_content);

        }
    }

    public CommentAdapter(List<Comment> commentList){
        mCommentList = commentList;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.comments_item,parent,false);
        final ViewHolder holder = new ViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {

        Comment comment = mCommentList.get(position);
        holder.commentatorName.setText(comment.getUser().getUsername());
        holder.commentatorContent.setText(comment.getContent());

    }

    @Override
    public int getItemCount() {
        return mCommentList.size();
    }
}

File: app/src/main/java/me/zhaoweihao/hnuplus/Adapter/CommentAdapter.kt
package me.zhaoweihao.hnuplus.Adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import butterknife.BindView
import butterknife.ButterKnife
import me.zhaoweihao.hnuplus.JavaBean.Comment
import me.zhaoweihao.hnuplus.R

/**
 * Created by Administrator on 2017/11/10.
 */

class CommentAdapter(private val mCommentList: List<Comment>) : RecyclerView.Adapter<CommentAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var commentatorName: TextView = view.findViewById(R.id.tv_commentator_name) as TextView
        var commentatorContent: TextView = view.findViewById(R.id.tv_commentator_content) as TextView
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.comments_item, parent, false)
        val holder = ViewHolder(view)
        return holder
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val comment = mCommentList[position]
        holder.commentatorName.text = comment.user!!.username
        holder.commentatorContent.text = comment.content
    }

    override fun getItemCount(): Int {
        return mCommentList.size
    }
}
